server/src/test/java/org/elasticsearch/index/mapper/ObjectMapperMergeTests.java

```diff
@@ -97,6 +108,18 @@ public void testMergeEnabledForRootMapper() {
         assertFalse(result.isEnabled());
     }
 
+    public void testMergeDisabledRootMapper() {
+        String type = MapperService.SINGLE_MAPPING_NAME;
+        final RootObjectMapper rootObjectMapper =
+            (RootObjectMapper) new RootObjectMapper.Builder(type, Version.CURRENT).enabled(false).build(new ContentPath());
+        //the root is disabled, and we are not trying to re-enable it, but we do want to be able to add runtime fields
+        final RootObjectMapper mergeWith =
+            new RootObjectMapper.Builder(type, Version.CURRENT).addRuntime(new TestRuntimeField("test", "long")).build(new ContentPath());
+
+        RootObjectMapper merged = (RootObjectMapper) rootObjectMapper.merge(mergeWith);
+        assertFalse(merged.isEnabled());
+    }
```

Comment:
Can we assert that the runtime field is still available as well?

Reply:
good point, I forgot that